apps/sms/js/contacts.js

```diff
@@ -206,31 +200,27 @@
           results = contacts;
         }
 
-        callback(results, {
-          terms: terms
-        });
-      };
-
-      request.onerror = function onerror() {
-        // When an error occurs, regardless of completed count,
-        // clear the onsuccess handler from this request and immediately
-        // invoke the callback with a `null` argument.
-        this.onsuccess = this.onerror = null;
-        callback(null);
-      };
+        return results;
+      }).catch(() => {
+        return null;
+      });
     },
-    findByString: function contacts_findBy(filterValue, callback) {
+
+    findByString: function contacts_findBy(filterValue) {
       var unknownCallback = function(contacts) {
         contacts = contacts || [];
-        this.findByUnknown(filterValue, function(unknown) {
-          unknown = unknown || [];
-          callback(contacts.concat(unknown));
-        });
+        return this.findByUnknown(filterValue)
```

Comment:
Maybe I'm biased, but if we use ```Promise.all``` to run this _find_ in "parallel" the code will look more readable:
```
return Promise.all([
  this.findContactByString(filterValue),
  this.findByUnknown(filterValue)
]).then(function(contacts) {
  var knownContacts = contacts[0];
  var unknownContacts = contacts[1];

  return unknownContacts.length > 0 ?
    knownContacts.concat(unknownContacts) :
    knownContacts;
});
```
What do you think?